### PR TITLE
Task1 Changes

### PR DESCRIPTION
**Pull request by bharathcn**
Please review the code, I have tried to implement below concepts 

- React.createElement

- React.Component

- React.PureComponent

- Functional component

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,23 +1,9 @@
-import logo from './logo.svg';
-import './App.css';
-
 function App() {
```
Comment (abash26) [inline src/App.js]: You can also use arrow function. It's preferable, have shorter syntax. e.g. 
```suggestion
const App = () => <div><p>Example of Functional Component.</p></div>
```

File: src/index.js [hunk 2]
```diff
@@ -2,16 +2,17 @@ import React from 'react';
...
 import App from './App';
+import Parent from './ParentComponent';
 import reportWebVitals from './reportWebVitals';
 
 ReactDOM.render(
+  <React.Fragment>
   <React.StrictMode>
     <App />
-  </React.StrictMode>,
+    <Parent/>
```
Comment (abash26) [inline src/index.js]: You should pass only 1 element in ReactDOM.render(**element**, container[, callback])
Parent should be inside App component.
it should look like:

    ReactDOM.render(<App />, document.getElementById('root'))

https://reactjs.org/docs/react-dom.html#reference

File: src/index.js [hunk 3]
```diff
@@ -2,16 +2,17 @@ import React from 'react';
...
+  </React.StrictMode>
```
Comment (abash26) [inline src/index.js]: StrictMode you can add in App component itself

File: src/ParentComponent.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import PureComponent from "./PureComponent";
+import ClassComponent from "./ClassComponent";
+import React from "react";
```
Comment (abash26) [inline src/ParentComponent.js]: I would suggest to use destructure here 
```suggestion
import { Component } from "react";
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#object_destructuring

File: src/index.js [hunk 5]
```diff
@@ -2,16 +2,17 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import './index.css';
 import App from './App';
+import Parent from './ParentComponent';
 import reportWebVitals from './reportWebVitals';
 
 ReactDOM.render(
+  <React.Fragment>
```
Comment (abash26) [inline src/index.js]: <React.Fragment> should be removed from here. It also have a short syntax <> </>

Comment (bharathcn) [reply src/index.js]: Changes are done.

File: src/ParentComponent.js (hunk 4 above)
Comment (bharathcn) [reply src/ParentComponent.js]: Changes are done

File: src/index.js (hunk 3 above)
Comment (bharathcn) [reply src/index.js]: Changes are done.

File: src/index.js (hunk 2 above)
Comment (bharathcn) [reply src/index.js]: Changes are done.

File: src/App.js (hunk 1 above)
Comment (bharathcn) [reply src/App.js]: Changes are done.

**Pull request merged**